src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -3194,6 +3226,22 @@ unsafe static void Main()
                 //         delegate*<C, void> ptr = &IHelpers.M;
                 Diagnostic(ErrorCode.ERR_AmbigCall, "IHelpers.M").WithArguments("IHelpers.M(I1)", "IHelpers.M(I2)").WithLocation(13, 35)
             );
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var addressOf = syntaxTree.GetRoot().DescendantNodes().OfType<PrefixUnaryExpressionSyntax>().Single();
+
+            string[] expectedSymbolCandidates = new[] { "void IHelpers.M(I1 i1)", "void IHelpers.M(I2 i2)" };
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, addressOf,
+                expectedSyntax: "&IHelpers.M",
+                expectedType: null,
+                expectedConvertedType: "delegate*<C, System.Void>",
+                expectedCandidateReason: CandidateReason.OverloadResolutionFailure,
+                expectedSymbolCandidates: expectedSymbolCandidates);
+
+            AssertEx.Equal(expectedSymbolCandidates, model.GetMemberGroup(addressOf).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
+            AssertEx.Equal(expectedSymbolCandidates, model.GetMemberGroup(addressOf.Operand).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
         }
 
         [Fact]
```

Comment:
I've removed these asserts as `VerifyFunctionPointerSemantics` now handles the `AddressOf` case.